### PR TITLE
Hw2 task3

### PR DESCRIPTION
**Pull request by E1-Bart0**
HomeWork2 Task3 Done

### CODE REVIEW

File: All_home_works/hw2/task3.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+def recursion(res, nums, args, index=0):
+    for num in args[index]:
+        nums.append(num)
+        if len(args) == index + 1:
+            res.append(nums.copy())
+        else:
+            recursion(res, nums, args, index=index + 1)
+        nums.pop()
+    return res
```
Comment (Cjkjvfnby) [inline All_home_works/hw2/task3.py]: Why do you need to return it?

File: All_home_works/hw2/task3.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    [1, 4],
+    [2, 3],
+    [2, 4],
+]
+"""
+from typing import Any, List
+
+
+def combinations(*args: List[Any]) -> List[List]:
+    res, nums = [], []
```
Comment (Cjkjvfnby) [inline All_home_works/hw2/task3.py]: `res` is usually named `accumulator` in the recursion context.

File: All_home_works/hw2/task3.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if args:
+        recursion(res, nums, args)
+    return res
+
+
+def recursion(res, nums, args, index=0):
```
Comment (Cjkjvfnby) [inline All_home_works/hw2/task3.py]: Could we structure this program in a way when you avoid passing `nums` as an argument?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
What is the limitation for the recursion in Python?

File: All_home_works/hw2/task3.py (hunk 1 above)
Comment (E1-Bart0) [reply All_home_works/hw2/task3.py]: As the lecturer said, a limitation for the recursion is something near 3000.
Of course, not needed to return a result. I cut it.

File: All_home_works/hw2/task3.py (hunk 3 above)
Comment (E1-Bart0) [reply All_home_works/hw2/task3.py]: No problem!

File: All_home_works/hw2/task3.py (hunk 1 above)
Comment (Cjkjvfnby) [reply All_home_works/hw2/task3.py]: Please lunch the console and check what is the limit on your Python.

Comment (E1-Bart0) [reply All_home_works/hw2/task3.py]: 987

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**